# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../libcsv/libcsv-3.0.0-fPIC.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Compile with -fPIC so it can be linked on 64 bit targets.

--- libcsv-3.0.0/Makefile.orig	2008-07-27 03:48:28.000000000 +0200
+++ libcsv-3.0.0/Makefile	2010-07-27 16:09:00.468180241 +0200
@@ -7,6 +7,8 @@
 INCDIR=/usr/include
 MANDIR=/usr/share/man/man3
 
+CFLAGS=-fPIC
+
 .PHONY: all install install_headers install_man install_shared install_static clean test
 
 all: libcsv.so libcsv.a
@@ -32,7 +34,7 @@
 	cp -f $< $(DESTDIR)$(LIBDIR)/$<
 
 libcsv.o: libcsv.c csv.h
-	$(CC) $< -c -o $@
+	$(CC) $(CFLAGS) $< -c -o $@
 
 libcsv.so: libcsv.o
 	$(CC) -shared $< -o $@
